### PR TITLE
[Feat/terraform] Add module

### PR DESCRIPTION
**Pull request by erdrix**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Add terraform modules to deploy NiFiKop on kubernetes cluster

### Why?

Part of a larger share on terraform modules 

### Checklist

- [X] Implementation tested

### CODE REVIEW

File: terraform/nifikop.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  metadata {
+    annotations = {
+      name = var.nifikop_namespace
+    }
+    name = var.nifikop_namespace
+  }
+}
+
+# Deploy CRDs
+resource "k8s_manifest" "nifikop_crds" {
```
Comment (michael81877) [inline terraform/nifikop.tf:13]: These will be automatically deployed along with the nifikop helm chart, so I don't believe it's necessary to deploy them alongside the chart. Unless this is just to be able to delete them from the Kubernetes cluster.

**Comment by michael81877** [comment]
Conceptual question about this: Do we intend to maintain a helm chart and a terraform plan for deploying nifikop & a nifi cluster to Kubernetes? Is this meant to be an example or kept up to date as the preferred way to deploy a complete setup?

File: terraform/nifikop.tf (hunk 1 above)
Comment (erdrix) [reply terraform/nifikop.tf:13]: It is necessary to update the CRDs, as helm is not able to update the existing CRDs, we have to deploy them explicitly in terraform, this way if we deploy a new version of CRDs they will be deployed.

**Comment by erdrix** [comment]
Not sure yet, for now in my mind the reference is the Helm chart for NiFiKop and the classic yaml for NiFi Cluster. 

But we are working on an application (`flow-manager`) to provide a user interface to manage the cluster and the data flow on top of NiFiKop. This application will require this terraform module and something else for the NiFi cluster (Helm Chart or a new high-level operator that will manage the NiFi cluster and other types of resources).

So no concrete answer for now, but what can lead the thinking could be this application

**Comment by michael81877** [comment]
Interesting. I wonder if this would belong in this repo vs another one for deployment-specific configurations? Here's why:

I think we'd likely use [ArgoCD ApplicationSets](https://argo-cd.readthedocs.io/en/stable/user-guide/application-set/) (see the various generators to parameterize deployments) to generate new deployments to accomplish a similar thing you've done here with a terraform module.  We do this today with nifikop, cert manager, and nifi-cluster helm chart (#141) deployments. The canonical ApplicationSet examples show deploying applications to many k8s clusters, but it could be many applications to one cluster just as well.

I suspect people will choose approaches that more readily adapt to their existing environment and so I wonder about maintaining stuff like that alongside the operator vs another repo? I don't know the correct answer. Just opening the dialog 🙂

**Comment by erdrix** [comment]
> Interesting. I wonder if this would belong in this repo vs another one for deployment-specific configurations? Here's why:
> 
> I think we'd likely use [ArgoCD ApplicationSets](https://argo-cd.readthedocs.io/en/stable/user-guide/application-set/) (see the various generators to parameterize deployments) to generate new deployments to accomplish a similar thing you've done here with a terraform module. We do this today with nifikop, cert manager, and nifi-cluster helm chart (#141) deployments. The canonical ApplicationSet examples show deploying applications to many k8s clusters, but it could be many applications to one cluster just as well.
> 
> I suspect people will choose approaches that more readily adapt to their existing environment and so I wonder about maintaining stuff like that alongside the operator vs another repo? I don't know the correct answer. Just opening the dialog 🙂

Thanks for the comment, I wasn't sure whether to put this terraform here or outside, I'm now convinced this is the wrong place :)

**Pull request closed without merging**